indexer/osm_editor.cpp

```diff
@@ -243,6 +243,35 @@ bool AreFeaturesEqualButStreet(FeatureType const & a, FeatureType const & b)
 
   return true;
 }
+
+XMLFeature GetMatchingFeatureFromOSM(osm::ChangesetWrapper & cw,
+                                     unique_ptr<FeatureType const> featurePtr)
+{
+  ASSERT(featurePtr->GetFeatureType() != feature::GEOM_LINE, ("Line features are not supported yet."));
+  if (featurePtr->GetFeatureType() == feature::GEOM_POINT)
+  {
+    return cw.GetMatchingNodeFeatureFromOSM(featurePtr->GetCenter());
+  }
+  else
+  {
+    // Set filters out duplicate points for closed ways or triangles' vertices.
+    set<m2::PointD> geometry;
+    featurePtr->ForEachTriangle([&geometry](m2::PointD const & p1,
+                                            m2::PointD const & p2, m2::PointD const & p3)
+    {
+      geometry.insert(p1);
```

Comment:
I'm not sure that OSM feature geometry also represented as a set of triangles. Why not to iterate over feature's points? @vng WDYT?